jsonschema-validation.md

```diff
@@ -320,115 +309,57 @@ can be used alongside the `type` keyword with a value of "integer", or could be
 is the recommended best practice.
 
-#### Format-Assertion Vocabulary
-
-When the Format-Assertion vocabulary is declared with a value of true,
-implementations MUST provide full validation support for all of the formats
-defined by this specification. Implementations that cannot provide full
-validation support MUST refuse to process the schema.
-
-An implementation that supports the Format-Assertion vocabulary:
-
-- MUST still collect `format` as an annotation if the implementation supports
-  annotation collection;
-- MUST evaluate `format` as an assertion;
-- MUST implement syntactic validation for all format attributes defined in this
-  specification, and for any additional format attributes that it recognizes,
-  such that there exist possible instance values of the correct type that will
-  fail validation.
-
 The requirement for minimal validation of format attributes is
 intentionally vague and permissive, due to the complexity involved in many of
 the attributes. Note in particular that the requirement is limited to syntactic
 checking; it is not to be expected that an implementation would send an email,
 attempt to connect to a URL, or otherwise check the existence of an entity
-identified by a format instance.[^4]
-
-[^4]: The expectation is that for simple formats such as date-time, syntactic
-validation will be thorough. For a complex format such as email addresses, which
-are the amalgamation of various standards and numerous adjustments over time,
-with obscure and/or obsolete rules that may or may not be restricted by other
-applications making use of the value, a minimal validation is sufficient. For
-example, an instance string that does not contain an "@" is clearly not a valid
-email address, and an "email" or "hostname" containing characters outside of
-7-bit ASCII is likewise clearly invalid.
-
-It is RECOMMENDED that implementations use a common parsing library for each
-format, or a well-known regular expression. Implementations SHOULD clearly
-document how and to what degree each format attribute is validated.
-
-The [standard core and validation meta-schema](#meta-schema) includes this
-vocabulary in its `$vocabulary` keyword with a value of false, since by default
-implementations are not required to support this keyword as an assertion.
-Supporting the format vocabulary with a value of true is understood to greatly
-increase code size and in some cases execution time, and will not be appropriate
-for all implementations.
+identified by a format instance.
 
 #### Custom format attributes
 
 Implementations MAY support custom format attributes. Save for agreement between
 parties, schema authors SHALL NOT expect a peer implementation to support such
-custom format attributes. An implementation MUST NOT fail to collect unknown
-formats as annotations. When the Format-Assertion vocabulary is specified,
-implementations MUST fail upon encountering unknown formats.
+custom format attributes.
```

Comment:
I think that given the changes in this section, we would always expect implementations to "support" custom format attributes, we just wouldn't expect them to validate to anything other than `true`. Since partial and even no validation is allowed for any format attribute, always returning `true` is a valid way to "support" that attribute.

I think the point of this sentence is to say that you can't expect it to assert the way you expect, but that ends up being the case for any format attribute including the ones defined in the spec. So, I'm not sure what we should do with this.

Reply:
The point might be moot with https://github.com/json-schema-org/json-schema-spec/issues/1520.

For the purpose here, though, I want to move away from specifying annotation or assertion behavior.  Maybe it's enough to say that for unknown formats, an implementation must collect the value as an annotation and provide a passing validation result.

Reply:
How does #1520 makes this irrelevant? Isn't that just about what the default behavior is? We still have multiple behavior that we have to define in consistent way.

> Maybe it's enough to say that for unknown formats, an implementation must collect the value as an annotation and provide a passing validation result.

That would work.

Reply:
Maybe [this](https://github.com/json-schema-org/json-schema-spec/pull/1510#discussion_r1648195159) is the solution here, too?